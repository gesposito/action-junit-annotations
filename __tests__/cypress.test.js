const { failingCasesFrom, annotationsFrom, mergeReports } = require("../index");

describe("Nested Cypress report", () => {
  let failingCases;
  test("Converts JUnit XML to failing cases in JSON", async () => {
    failingCases = await failingCasesFrom(
      "./test-results/cypress-junit-failure.xml"
    );
    expect(failingCases).toMatchSnapshot();
  });

  test("Converts failing cases in JSON to GitHub annotations", async () => {
    const annotations = annotationsFrom(failingCases);
    expect(annotations).toMatchSnapshot();
  });
});

describe("Multiple Cypress reports", () => {
  let failingCases;
  test("Converts JUnit XML to failing cases in JSON", async () => {
    const reportPath = "./test-results/cypress-results/";
    const mergedPath = await mergeReports(reportPath);
    failingCases = await failingCasesFrom(mergedPath);
    expect(failingCases).toMatchSnapshot();
  });

  test("Converts failing cases in JSON to GitHub annotations", async () => {
    const annotations = annotationsFrom(failingCases);
    expect(annotations).toMatchSnapshot();
  });
});
