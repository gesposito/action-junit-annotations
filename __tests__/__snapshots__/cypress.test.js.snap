// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Cypress reports Converts JUnit XML to failing cases in JSON 1`] = `
Array [
  Object {
    "classname": "By visiting google.com, page contains GitHub",
    "failure": Object {
      "#text": "AssertionError: Timed out retrying: Expected to find content: 'GitHub' but never did.
    at Context.eval (https://www.google.com/__cypress/tests?p=cypress/integration/github_spec.js:97:130)",
      "message": "Timed out retrying: Expected to find content: 'GitHub' but never did.",
      "type": "AssertionError",
    },
    "name": "GitHub By visiting google.com, page contains GitHub",
    "path": "GitHub By visiting google.com, page contains GitHub",
    "time": "5.9800",
  },
  Object {
    "classname": "By visiting google.com, title contains GitHub",
    "failure": Object {
      "#text": "AssertionError: Timed out retrying: expected '<title>' to contain 'GitHub'
    at Context.eval (https://www.google.com/__cypress/tests?p=cypress/integration/github_spec.js:97:235)",
      "message": "Timed out retrying: expected '&lt;title>' to contain 'GitHub'",
      "type": "AssertionError",
    },
    "name": "GitHub By visiting google.com, title contains GitHub",
    "path": "GitHub By visiting google.com, title contains GitHub",
    "time": "4.3910",
  },
]
`;

exports[`Multiple Cypress reports Converts failing cases in JSON to GitHub annotations 1`] = `
Array [
  Object {
    "annotation_level": "failure",
    "end_line": 0,
    "message": "AssertionError: Timed out retrying: Expected to find content: 'GitHub' but never did.
    at Context.eval (https://www.google.com/__cypress/tests?p=cypress/integration/github_spec.js:97:130)",
    "path": "GitHub By visiting google.com, page contains GitHub",
    "start_line": 0,
  },
  Object {
    "annotation_level": "failure",
    "end_line": 0,
    "message": "AssertionError: Timed out retrying: expected '<title>' to contain 'GitHub'
    at Context.eval (https://www.google.com/__cypress/tests?p=cypress/integration/github_spec.js:97:235)",
    "path": "GitHub By visiting google.com, title contains GitHub",
    "start_line": 0,
  },
]
`;

exports[`Nested Cypress report Converts JUnit XML to failing cases in JSON 1`] = `
Array [
  Object {
    "classname": "By visiting google.com, page contains GitHub",
    "failure": Object {
      "#text": "AssertionError: Timed out retrying: Expected to find content: 'GitHub' but never did.
    at Context.eval (https://www.google.com/__cypress/tests?p=cypress/integration/index_spec.js:97:367)",
      "message": "Timed out retrying: Expected to find content: &apos;GitHub&apos; but never did.",
      "type": "AssertionError",
    },
    "name": "Internet By visiting google.com, page contains GitHub",
    "path": "Internet By visiting google.com, page contains GitHub",
    "time": "5.8030",
  },
  Object {
    "classname": "By visiting google.com, title contains GitHub",
    "failure": Object {
      "#text": "AssertionError: Timed out retrying: expected '<title>' to contain 'GitHub'
    at Context.eval (https://www.google.com/__cypress/tests?p=cypress/integration/index_spec.js:97:472)",
      "message": "Timed out retrying: expected &apos;&lt;title&gt;&apos; to contain &apos;GitHub&apos;",
      "type": "AssertionError",
    },
    "name": "Internet By visiting google.com, title contains GitHub",
    "path": "Internet By visiting google.com, title contains GitHub",
    "time": "4.3790",
  },
]
`;

exports[`Nested Cypress report Converts failing cases in JSON to GitHub annotations 1`] = `
Array [
  Object {
    "annotation_level": "failure",
    "end_line": 0,
    "message": "AssertionError: Timed out retrying: Expected to find content: 'GitHub' but never did.
    at Context.eval (https://www.google.com/__cypress/tests?p=cypress/integration/index_spec.js:97:367)",
    "path": "Internet By visiting google.com, page contains GitHub",
    "start_line": 0,
  },
  Object {
    "annotation_level": "failure",
    "end_line": 0,
    "message": "AssertionError: Timed out retrying: expected '<title>' to contain 'GitHub'
    at Context.eval (https://www.google.com/__cypress/tests?p=cypress/integration/index_spec.js:97:472)",
    "path": "Internet By visiting google.com, title contains GitHub",
    "start_line": 0,
  },
]
`;
